lib/honeybadger/events_worker.rb

```diff
@@ -0,0 +1,329 @@
+      end
+
+      queue.push(msg)
+    end
+
+    def send_now(msg)
+      handle_response(send_to_backend(msg))
+    end
+
+    def shutdown(force = false)
+      d { 'shutting down worker' }
+
+      mutex.synchronize do
+        @shutdown = true
+      end
+
+      return true if force
+      return true unless thread&.alive?
+
+      if throttled?
+        warn { sprintf('Unable to send %s event(s) to Honeybadger (currently throttled)', queue.size) } unless queue.empty?
+        return true
+      end
+
+      info { sprintf('Waiting to send %s events(s) to Honeybadger', queue.size) } unless queue.empty?
+      queue.push(FLUSH)
+      queue.push(SHUTDOWN)
+      !!thread.join
+    ensure
+      queue.clear
+      kill!
+    end
+
+    # Blocks until queue is processed up to this point in time.
+    def flush
+      mutex.synchronize do
+        if thread && thread.alive?
+          queue.push(FLUSH)
+          queue.push(marker)
+          marker.wait(mutex)
+        end
+      end
+    end
+
+    def start
+      return false unless can_start?
+
+      mutex.synchronize do
+        @shutdown = false
+        @start_at = nil
+
+        return true if thread&.alive?
+
+        @pid = Process.pid
+        @thread = Thread.new { run }
+        @timeout_thread = Thread.new { schedule_timeout_check }
+      end
+
+      true
+    end
+
+    private
+
+    attr_reader :config, :queue, :pid, :mutex, :marker, :thread, :timeout_thread, :throttle,
+      :throttle_interval, :start_at, :send_queue, :last_sent, :max_events, :send_timeout
+
+    def_delegator :config, :backend
+
+    def shutdown?
+      mutex.synchronize { @shutdown }
+    end
+
+    def suspended?
+      mutex.synchronize { start_at && Time.now.to_i < start_at }
+    end
+
+    def can_start?
+      return false if shutdown?
+      return false if suspended?
+      true
+    end
+
+    def throttled?
+      mutex.synchronize { throttle > 0 }
+    end
+
+    def kill!
+      d { 'killing worker thread' }
+
+      if thread
+        Thread.kill(thread)
+        Thread.kill(timeout_thread)
+        thread.join # Allow ensure blocks to execute.
+      end
+
+      true
+    end
+
+    def suspend(interval)
+      mutex.synchronize do
+        @start_at = Time.now.to_i + interval
+        queue.clear
+      end
+
+      # Must be performed last since this may kill the current thread.
+      kill!
+    end
+
+    def schedule_timeout_check
+      loop do
+        sleep(send_timeout / 1000.0)
+        ms_since = (Time.now.to_f - last_sent.to_f) * 1000.0
+        if ms_since >= send_timeout
+          queue.push(CHECK_TIMEOUT)
+        end
+      end
+    end
+
+    def run
+      begin
+        d { 'worker started' }
+        mutex.synchronize do
+          @last_sent = Time.now
+        end
+        loop do
+          # ms_since = (Time.now.to_f - @last_sent.to_f) * 1000.0
+          # if ms_since >= send_timeout
+          #   queue.push(CHECK_TIMEOUT)
+          # end
```

Comment:
```suggestion     
```

Reply:
Is this leftover or unfinished code? Can we remove if not necessary?